consensus/misc/eip1559.go

```diff
@@ -52,21 +52,22 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header) *big.Int {
 		return new(big.Int).SetUint64(params.InitialBaseFee)
 	}
 
-	// If the parent gasUsed is the same as the target, the baseFee remains unchanged.
-	if parent.GasUsed == parent.GasLimit {
-		return new(big.Int).Set(parent.BaseFee)
-	}
-
 	var (
-		gasLimit                 = new(big.Int).SetUint64(parent.GasLimit)
+		gasTarget                = parent.GasLimit / params.ElasticityMultiplier
+		gasTargetBig             = new(big.Int).SetUint64(gasTarget)
```

Comment:
If this is the fork block, then here we should not divide by `params.ElasticityMultiplier`, I think.

Reply:
@holiman actually I am now realizing this isn't an issue, because if we go with the behaviour where on the fork block we return the `InitialBaseFee` parameter and do no calculation. Since the EIP-1559 spec didn't have this behaviour yet, we needed to update the calculation logic to not accidentally halve the `gasTarget`. I am going to update the spec now though to match the client behaviour and this issue should disappear.

Reply:
It is still an issue with the `gasTarget` increase though as you noted below. But I think the `CalcBaseFee` is correct.